### PR TITLE
Fixes #18105: Implement case within variable declaration

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/18105

### CODE REVIEW

File: rudder-lang/src/ast.rs [hunk 1]
```diff
@@ -179,16 +180,37 @@ impl<'src> AST<'src> {
...
+                        Ok(()) => { self.variable_definitions.insert(name, val); }
+                    }
                 }
             }
         }
     }
 
+    /// Add the variables extensions to the global definition space
+    fn add_variable_extensions(&mut self, variables: Vec<PVariableExt<'src>>) {
+        let context = Rc::get_mut(&mut self.context).expect("Context has not been allocated before variables !");
```
Comment (gpoblon) [inline rudder-lang/src/ast.rs:198]: don;t you want to use it ?

Comment (peckpeck) [reply rudder-lang/src/ast.rs:198]: oops, useless, indeed

**Comment by peckpeck** [comment]
PR updated with a new commit

**Review by gpoblon — APPROVED** [review]

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**